bin/start-vm

```diff
@@ -1,523 +1,728 @@
+        # Homebrew path on macOS
+        if [ "$os" = macos ]; then
+            # Test for HomeBrew path
+            homebrew_prefix=$(brew info qemu | awk 'NR==4{print $1}')
+            uefi_code="${uefi_code:-$homebrew_prefix/share/qemu/edk2-aarch64-code.fd}"
+            uefi_vars="${uefi_vars:-$homebrew_prefix/share/qemu/edk2-arm-vars.fd}"
+        elif [ "$os" = debian ]; then
+            uefi_code="${uefi_code:-/usr/share/AAVMF/AAVMF_CODE.fd}"
+            uefi_vars="${uefi_vars:-/usr/share/AAVMF/AAVMF_VARS.fd}"
+        elif [ "$os" = centos ]; then
+            # Note: Running `AARCH64` on `x86_64` requires `qemu-system-aarch64`
+            # package which is not present in official repositories
+            uefi_code="${uefi_code:-/usr/share/AAVMF/AAVMF_CODE.fd}"
+            uefi_vars="${uefi_vars:-/usr/share/AAVMF/AAVMF_VARS.fd}"
+        else
+            echo "Error: Could not find UEFI code and vars file."
+            exit 1
+        fi
 
     fi
 
-fi
+    # Define UEFI options for QEMU
+    if [ $uefi ]; then
 
+        [ -r $uefi_code ] || eusage "Missing uefi code at $uefi_code.\n Run: apt-get install ovmf"
+        [ -r $uefi_vars ] || eusage "Missing uefi vars at $uefi_vars.\n Run: apt-get install ovmf"
 
-# Validate AARCH64/ARM64 arch
-if [ "$arch" = aarch64 ]; then
+        [ -e $TARGET_DIR/$mac.vars ] || cp $uefi_vars $TARGET_DIR/$mac.vars
+
+        # Add support for amd64 architecture
+        if [ "$arch" = amd64 ]; then
+            QEMU_OPTS+=("-global driver=cfi.pflash01,property=secure,value=on")
+        fi
+
+        # Set UEFI code and vars files
+	    QEMU_OPTS+=("-drive if=pflash,format=raw,unit=0,file=${uefi_code},readonly=on"
+                    "-drive if=pflash,format=raw,unit=1,file=$TARGET_DIR/$mac.vars")
 
-    # Evaluate possible ARM64 arch outputs
-    testArch="undefined"
-    if [ "arm64" = $(uname -m) ]; then
-        testArch="aarch64"
-    fi
-    if [ "aarch64" = $(uname -m) ]; then
-        testArch="aarch64"
     fi
+}
 
-    # Validate for native or emulated arch
-    if [ "$arch" = "$testArch" ]; then
 
-        # Running on native arch
-        # Check for acceleration support
-        if [ "$running_os" = macos ]; then
-            # macOS HVF support is always given on all new Intel Macs
-            # that are supported by the current HomeBrew version
-            virtOpts+=("-cpu host" "-machine virt" "-accel hvf")
+# Validate and set the CPU architecture
+# cores and chipsets
+qemu_opt_cpu () {
+    # Set CPU count
+    QEMU_OPTS+=("-smp $cpu")
 
-        else
+    # Validate amd64 arch
+    if [ "$arch" = amd64 ]; then
+
+       # Evaluate possible AMD64 arch outputs
+        test_arch="undefined"
+
+        if [ "amd64" = $(uname -m) ]; then
+            test_arch="amd64"
+        fi
+
+        if [ "x86_64" = $(uname -m) ]; then
+            test_arch="amd64"
+        fi
+
+        # Validate for native or emulated arch
+        if [ "$arch" = "$test_arch" ]; then
```

Comment:
By using `get_arch.sh` here, you don't need to define a temp variable like `test_arch` and could simply compare `$arch` with the result of `get_arch.sh`

Reply:
This wouldn't offer legacy support when someone defines `--arch` for emulating as `x86_64` or `aarch64` and would only work if the user types `amd64` or `arm64` which is uncommon. Therefore, we need to validate this before which is stored in a var called `test_arch`.

Reply:
You already normalize the `arch` in `set_binaries`, so `arch` can only be `arm64` or `amd64` at this point. I would also suggest to create a new function called `normalize_arch` which takes the if statement of `set_binaries` and executes them right after `get_opts` (or make it a part of `get_opts`)